codex/blockexchange/engine/engine.nim

```diff
@@ -529,8 +529,13 @@ proc taskHandler*(b: BlockExcEngine, task: BlockExcPeerCtx) {.gcsafe, async.} =
 
   trace "wantsBlocks", peer = task.id, n = wantsBlocks.len
   if wantsBlocks.len > 0:
-    trace "Got peer want blocks list", items = wantsBlocks.len
-
+    # Consider only the wants not currently in-flight,
+    # Then mark them as in-flight.
+    wantsBlocks.keepItIf(not it.inFlight)
```

Comment:
You probably just want to only get the non-inflight in the `filterIt` above, instead of the `keepItIf`, like so:

```nim
  var
    wantsBlocks = task.peerWants.filterIt(
      it.wantType == WantType.WantBlock and not it.inFlight
    )
```